include/deal.II/lac/constraint_matrix.templates.h

```diff
@@ -670,7 +682,7 @@ distribute_local_to_global (const Vector<LocalType>      &local_vector,
                 continue;
 
               const ConstraintLine &position_j =
-                lines[lines_cache[calculate_line_index(local_dof_indices[j])]];
+                lines[lines_cache[calculate_line_index(local_dof_indices_row[j])]];
```

Comment:
Scratch this comment - what I describe is exactly what you. I must have missed the forest for the trees.